### PR TITLE
MMCA 4933 - Add 'maxTransactionsExceeded' field to responseDetail section

### PR DESCRIPTION
**Pull request by HariHmrc**

### CODE REVIEW

**Review by AdamCoops — DISMISSED** [review]

File: test/services/CashTransactionsServiceSpec.scala [hunk 1]
```diff
@@ -92,10 +92,71 @@ class CashTransactionsServiceSpec extends SpecBase {
...
+        result mustBe Right(expectedResult)
+        result match {
+          case Right(cashTransactions) => cashTransactions.maxTransactionsExceeded mustBe Some(false)
+          case Left(errorRsp) => fail(s"Expected CashTransactions but got ErrorResponse : $errorRsp")
+        }
+      }
+    }
+
+    "return Right with the Cash transactions on a successful response from the API " +
+      "with maxTransactionsExceeded Flag" in new Setup {
```
Comment (tyagikumarvineet) [inline test/services/CashTransactionsServiceSpec.scala:108]: better to say maxTransactionsExceeded as true

File: test/services/CashTransactionsServiceSpec.scala [hunk 2]
```diff
@@ -105,7 +166,7 @@ class CashTransactionsServiceSpec extends SpecBase {
       )
       running(app) {
         val result = await(service.retrieveCashTransactionsSummary("can", dateFrom, dateTo))
-        val expectedResult = CashTransactions(Nil, Nil)
+        val expectedResult = CashTransactions(Nil, Nil, None)
```
Comment (tyagikumarvineet) [inline test/services/CashTransactionsServiceSpec.scala:169]: Please remove None as the third parameter as it is already defaulted to None. Same as line 257

File: test/services/DomainServiceSpec.scala [hunk 3]
```diff
@@ -55,6 +57,23 @@ class DomainServiceSpec extends SpecBase {
...
+      result.maxTransactionsExceeded mustBe Some(true)
+      result.pendingTransactions.size mustBe 1
+      result.pendingTransactions.head.movementReferenceNumber mustBe "someId"
+      result.pendingTransactions.head.declarantEori mustBe EORI("someEori")
+      result.pendingTransactions.head.declarantReference mustBe Some("reference")
+      result.cashDailyStatements.size mustBe 1
+      result.cashDailyStatements.head.date mustBe dateFrom.toString
+      result.cashDailyStatements.head.declarations.size mustBe 1
+      result.cashDailyStatements.head.openingBalance mustBe "10000"
+
```
Comment (tyagikumarvineet) [inline test/services/DomainServiceSpec.scala:74]: Please remove this redundant line break.

File: app/domain/CashDailyStatement.scala [hunk 4]
```diff
@@ -19,7 +19,9 @@ package domain
 import models.EORI
 import play.api.libs.json.{Json, OFormat}
 
-case class CashTransactions(pendingTransactions: Seq[Declaration], cashDailyStatements: Seq[CashDailyStatement])
+case class CashTransactions(pendingTransactions: Seq[Declaration],
```
Comment (tyagikumarvineet) [inline app/domain/CashDailyStatement.scala:22]: Could you please write a test case for Json writes at least?

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**